File: backend/src/routes/notifications.js
import express from 'express';
import authMiddleware from '../middlewares/auth.js';
import { supabase } from '../services/supabaseClient.js';
const router = express.Router();
router.use(authMiddleware);
router.get('/', async (req, res, next) => {
    try {
        const userId = req.user.id;
        const { data, error } = await supabase.from('notifications').select('*').eq('user_id', userId).order('created_at', { ascending: false });
        if (error)
            return res.status(400).json({ error: error.message });
        res.json(data);
    }
    catch (err) {
        next(err);
    }
});
router.post('/mark-read', async (req, res, next) => {
    try {
        const userId = req.user.id;
        await supabase.from('notifications').update({ read: true }).eq('user_id', userId).eq('read', false);
        res.json({ message: 'All notifications marked as read.' });
    }
    catch (err) {
        next(err);
    }
});
export default router;

File: backend/routers/notifications.py
from fastapi import APIRouter, Depends
from ..schemas import User
from ..dependencies import get_current_user
from ..mock_db import mock_storage

router = APIRouter()

@router.get("/")
def get_notifications(user: User = Depends(get_current_user)):
    user_notifications = [n for n in mock_storage["notifications"] if n.get("user_id") == user.id]
    return sorted(user_notifications, key=lambda x: x.get("created_at", ""), reverse=True)

@router.post("/mark-read")
def mark_read(user: User = Depends(get_current_user)):
    for notification in mock_storage["notifications"]:
        if notification.get("user_id") == user.id and not notification.get("read"):
            notification["read"] = True
    return {"message": "All notifications marked as read."}
